src/app_service/common/message.nim

```diff
@@ -12,6 +12,7 @@ proc replacePubKeysWithDisplayNames*(allKnownContacts: seq[ContactsDto], message
     updatedMessage = updatedMessage.replaceWord(pair[1], pair[0])
 
   for pk in pubKeys:
+    let pk = pk # TODO https://github.com/nim-lang/Nim/issues/16740
```

Comment:
Do we need this workaround for every loop we do? It seems like no because you didn't change them all, so why those in particular?

Reply:
Only those that capture the loop variable in a closure (`.. => ..` creates a closure) - in _most_ cases, there's a corresponding `xxxIt` that uses a template and inlining instead - `allKnownContacts.filter(x => "@" & x.id == pk)` can be written as `allKnownContacts.filterIt(it.id == pk)` instead.